core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodec.java

```diff
@@ -57,12 +58,12 @@ public DataType getCqlType() {
   @Nullable
   @Override
   public ByteBuffer encode(
-      @Nullable CqlVector<SubtypeT> value, @NonNull ProtocolVersion protocolVersion) {
+      @Nullable List<SubtypeT> value, @NonNull ProtocolVersion protocolVersion) {
     if (value == null || cqlType.getDimensions() <= 0) {
       return null;
     }
     ByteBuffer[] valueBuffs = new ByteBuffer[cqlType.getDimensions()];
-    Iterator<SubtypeT> values = value.getValues().iterator();
+    Iterator<SubtypeT> values = value.iterator();
     int allValueBuffsSize = 0;
     for (int i = 0; i < cqlType.getDimensions(); ++i) {
       ByteBuffer valueBuff = this.subtypeCodec.encode(values.next(), protocolVersion);
```

Comment:
ListCodec and MapCodec have some handling around null elements and classcastexceptions, do we want to have the same checks here?

Reply:
My original thought was that this wasn't necessary because the codecs were pretty good about not meeting these error cases.  Then I went back and looked at the code and realized that recollection was entirely wrong.  Added that code in there (as well as check for a few other error conditions) and beefed up the codec test a bit to detect them.